src/Request/ReportRequest.php

```diff
@@ -0,0 +1,162 @@
+        'paid',
+        'all',
+        'settled'
+    ];
+
+    private $requestType;
+    private $callbackUrl;
+    private $paymentStatus;
+    private $startDate;
+    private $endDate;
+    private $limit;
+    private $reportFields;
+    private $subMerchant;
+
+    use ObjectPropertyConverter;
+
+    public function validate()
+    {
+        $props = get_object_vars($this);
+
+        if (empty($props['requestType'])) {
+            throw new ValidationException('RequestType can not be empty');
+        }
+
+        if (!in_array($props['requestType'], ['json', 'csv'])) {
+            throw new ValidationException('RequestType must be either "json" or "csv"');
+        }
+
+        if (empty($props['callbackUrl'])) {
+            throw new ValidationException('CallbackUrl can not be empty');
+        }
+
+        if ($props['paymentStatus'] && !in_array($props['paymentStatus'], self::PAYMENT_STATUSES)) {
+            throw new ValidationException('Invalid paymentStatus value');
+        }
+
+        if ($props['limit'] > 50000) {
+            throw new ValidationException('Limit exceeds maximum value of 50000');
+        }
+
+        if ($props['reportFields'] && is_array($props['reportFields'])) {
+            throw new ValidationException('ReportFields must be type of array');
+        }
+
+        return true;
+    }
+
+    /**
+     * Set request type.
+     *
+     * @param string $requestType
+     * @return $this
+     */
+    public function setRequestType(string $requestType): self
+    {
+        $this->requestType = $requestType;
+        return $this;
+    }
+
+    /**
+     * Set callback url.
+     *
+     * @param string $callbackUrl
+     * @return $this
+     */
+    public function setCallbackUrl(string $callbackUrl): self
+    {
+        $this->callbackUrl = $callbackUrl;
+        return $this;
+    }
+
+    /**
+     * Set payment statuses.
+     *
+     * @param string $paymentStatus
+     * @return $this
+     */
+    public function setPaymentStatus(string $paymentStatus): self
+    {
+        $this->paymentStatus = $paymentStatus;
+        return $this;
+    }
+
+    /**
+     * Set start date.
+     *
+     * @param string $startDate
+     * @return $this
+     */
+    public function setStartDate(string $startDate): self
+    {
+        $this->startDate = $startDate;
+        return $this;
+    }
+
+    /**
+     * Set end date.
+     *
+     * @param string $endDate
+     * @return $this
+     */
+    public function setEndDate(string $endDate): self
+    {
+        $this->endDate = $endDate;
+        return $this;
+    }
+
+    /**
+     * Set limit.
+     *
+     * @param int $limit
+     * @return $this
+     */
+    public function setLimit(int $limit): self
+    {
+        $this->limit = $limit;
+        return $this;
+    }
+
+    /**
+     * Set report fields.
+     *
+     * @param string $reportFields
+     * @return $this
+     */
+    public function setReportFields(string $reportFields): self
```

Comment:
`reportFields` should be an array of strings.

Reply:
Fixed in #44